### ISSUE TITLE
API endpoint is 404

### ISSUE THREAD

**Issue by lutzky** [pyalarmdotcom/alarmdotcom#158]
I'm using the integration with GoKonnect (which uses alarm.com), and it's giving this error:

> Retrying setup: 0, message='Attempt to decode JSON with unexpected mimetype: text/html; charset=utf-8', url=URL('https://www.alarm.com/web/api/video/cameras/')

Indeed, the URL is giving a 404 error (with a full HTML render). This means that:

1. The integration doesn't check for a 200 status
2. Perhaps more importantly, the API is gone 😱 

I'll research a bit more, please dedupe this against preexisting bugs if any.

**Comment by lutzky** [comment]
The official alarm.com web UI appears visually the same as before. It appears that the requests are going to  https://www.alarm.com/web/Video/SettingsMain_V2.aspx.

The issue is perhaps better placed at https://github.com/pyalarmdotcom/pyalarmdotcomajax, but I think this repo would have better visibility.

**Comment by lutzky** [comment]
Hmm, that's probably not quite it - the calls were being sent to `SettingsMain_V2` already back in the first comment on #122. Is it possible that there is a "new API", and "old API", and the "old API" has just been turned down?

Other important information: The integration was working fine until 10:59:55 UTC, and has been down since. That seems like a suspiciously round timestamp to stop working at.

**Comment by bkr1969** [comment]
Same here.  Not sure why there is anything to do with a camera as I do not have one.
```
Retrying setup: 0, message='Attempt to decode JSON with unexpected mimetype: text/html; charset=utf-8', url=URL('https://www.alarm.com/web/api/video/cameras/')
```

**Comment by lutzky** [comment]
This reproduces nicely with just `adc get` as well, same exception.

**Comment by Anto79-ops** [comment]
Hi, also noticed the message in my logs this morning, but I was running  0.2.10 until yesterday (and didn't see this message), so not sure if its related to 2.12 or something broke in the back-end.

```
This error originated from a custom integration.

Logger: custom_components.alarmdotcom.alarmhub
Source: custom_components/alarmdotcom/alarmhub.py:240
Integration: Alarm.com (documentation, issues)
First occurred: 4:59:59 AM (199 occurrences)
Last logged: 8:17:59 AM

custom_components.alarmdotcom.alarmhub: Update failed: 0, message='Attempt to decode JSON with unexpected mimetype: text/html; charset=utf-8', url=URL('https://www.alarm.com/web/api/video/cameras/')
```

**Comment by Anto79-ops** [comment]
elahd is really good and responsive, he's probably on vacation and most of these awesome developers do this for free as a side job. I'm sure he'll get at it when he can.

**Comment by jmutnick** [comment]
I'm having the same issue.  Following.

**Comment by lutzky** [comment]
Yep, let's try to help him narrow it down while we're out. Specifically, I believe I have a workaround:

```patch
diff --git a/pyalarmdotcomajax/__init__.py b/pyalarmdotcomajax/__init__.py
index 446e4ad..5f8dac8 100644
--- a/pyalarmdotcomajax/__init__.py
+++ b/pyalarmdotcomajax/__init__.py
@@ -537,6 +537,10 @@ class AlarmController:
         """Get data for the specified device types and build objects."""
 
         for device_type in device_types:
+            log.info('Operating on device_type %s', device_type)
+            if device_type == DeviceType.CAMERA:
+                log.info("It's a camera, skipping")
+                continue
             #
             # DETERMINE DEVICE'S PYALARMDOTCOMAJAX PYTHON CLASS
             #
```

That, at least, gets the `adc` command going again, indicating that the issue seems localized to cameras. I specifically *do* have a camera device, but at least this should get the alarm working.

**Comment by tholland15** [comment]
`https://www.alarm.com/web/api/video/cameras/` is totally gone or has changed to a new endpoint. 

@elahd **if** this endpoint has just changed, may I suggest that instead of just updating it to the new endpoint, can we instead fix it so that an existing endpoint going down does not nuke the entire component for all of us? Especially for those of us that do not have the devices in question on our system (in this case cameras).

**Comment by lutzky** [comment]
I can confirm that the workaround from https://github.com/pyalarmdotcom/alarmdotcom/issues/158#issuecomment-1208251609 works; unfortunately it accomplishes that by, indeed, disabling integration with the camera.

**Comment by bkr1969** [comment]
How can I implement this?  I didn't see a section with ```for device_type in device_types``` in my .py file.

**Comment by uvjustin** [comment]
I've been away from the integration for a long time and @elahd has basically taken over. I just pinged him about this, but if he's away for long or busy today I can push a temporary fix to disable cameras.

**Comment by cjones813** [comment]
I am also having the same issue. I was running 0.2.10 and noticed the error. Updated to 0.2.12 and error still continues

**Comment by cjeffcoat80** [comment]
I am having the same issue. Following

**Comment by uvjustin** [comment]
Just pushed v0.2.13 - see if it resolves things

**Comment by bkr1969** [comment]
Stupid question...how do we manually update?

**Comment by tholland15** [comment]
> Just pushed v0.2.13 - see if it resolves things

Fixed for me. Thanks!



> Stupid question...how do we manually update?

Go to the integration in HACS, hit the 3-dot in the upper right, then "Update Information" after that back out and HACS will show there is an update available.

**Comment by bkr1969** [comment]
No option like that.
![image](https://user-images.githubusercontent.com/42385989/183473959-70eb42c0-928f-4434-81cd-22b68ea3378c.png)

**Comment by bkr1969** [comment]
Isn't this set up directly in the UI now?  I don't even have it installed via HACS.

**Comment by uvjustin** [comment]
> Isn't this set up directly in the UI as an included integration now? I don't even have it installed via HACS.

Although the setup is now done in the config flow, it is still a custom component, and thus it requires putting files in the custom_components directory. If you don't use HACS, just download the archive from [here](https://github.com/pyalarmdotcom/alarmdotcom/releases/tag/v0.2.13) and put the files from the `custom_components/alarmdotcom` directory into your HA `custom_components/alarmdotcom` directory.

**Comment by bkr1969** [comment]
Weird.  I deleted my integration and installed via HACS.  Working now.  Thanks!

**Comment by cjeffcoat80** [comment]
works now. Thank you!!!

**Comment by Anto79-ops** [comment]
on the note above. Thanks looks like the issues are fixed....but how to enable or check if I have beta versions enabled? Here's what I see on the HACS end. Update information only gives me the latest release version

![image](https://user-images.githubusercontent.com/79593761/183489747-e3617c4a-1ef5-4df6-949a-7d5bd565f79e.png)

**Comment by elahd** [comment]
Thanks all for working through this and @uvjustin for stemming the leak.

First, if you're seeing the screen that @Anto79-ops posted above when you go to update Alarm.com in HACS, it's because the repo moved from uvjustin's namespace to the pyalarmdotcom GitHub org. You can reattach your instance by reinstalling from HACS. (You may need to delete the old version first.)

I can't replicate the issue on my end, but I pushed out a beta update that should:

1. Handle broken endpoints more gracefully (to @tholland15's point above).
2. Fix the issue with the camera endpoint so that people with Skybell HDs can continue to control those devices via this integration.

I also added tests to ensure that 404 errors on device endpoints don't break the entire flow.

This issue was caused by Alarm.com changing the URLs for a few of their camera endpoints. This breaks installs for accounts that don't have cameras because the integration scans all device categories to see if you have one of those devices.

I'd appreciate feedback on whether the fix works, both from people without cameras and people with Skybell HDs (looking at you, @lutzky).

The beta version is [v0.2.14-beta.0](https://github.com/pyalarmdotcom/alarmdotcom/releases/tag/v0.2.14-beta.0).

**Comment by Anto79-ops** [comment]
thanks for your quick response and fix! I actully had reinstalled it before (a couple of data ago, actually) so I was about to delete the integration and add it again today, BUT instead, I just went to "redownload", and clicked "enable beta" and I was able to select and download the beta version no problem and running it as we speak!

That page is still blank, so begs the question, should I still delete the integration and add it again? I mean this is minor, as long the code gets updated, having a blank screen for the integration information is not too concerning for me.

**Comment by elahd** [comment]
Weird, I see this:

![image](https://user-images.githubusercontent.com/466460/183505466-c6c5980b-cb4a-4a13-90ff-46bbf95e09d3.png)

Which version of HACS are you using?

![image](https://user-images.githubusercontent.com/466460/183505515-866df6f0-e75b-4ea1-abc4-da9b077d5c8b.png)

**Comment by cjones813** [comment]
![image](https://user-images.githubusercontent.com/57048835/183506211-e4d5db1a-6173-45e6-b669-8b40b9d5fb34.png) i can confirm I also see a blank page. I have installed the beta and its working great! Thank you!

**Comment by Anto79-ops** [comment]
yes the beta is also work great for me 

As for HACS, I have the same version.

![image](https://user-images.githubusercontent.com/79593761/183506493-3e97e1cb-97b9-469b-9bd2-fd79049f8f10.png)

**Comment by lutzky** [comment]
Thank you for the super fast turnaround on this! Works perfectly, other than the fact that the entity name for the chime has changed (from `switch.indoor_chime` to `switch.front_door_indoor_chime`). This might have happened quite a few versions ago, I hadn't updated in a while. 

Good to close, as far as I'm concerned. I'm also seeing "The developer has not provided any more information for this repository", in HACS 1.26.2, but that's perhaps for another bug.

**Comment by andrewheavin** [comment]
> Just pushed v0.2.13 - see if it resolves things

Worked for me. Thanks!

**Comment by mikesalz** [comment]
v0.2.13 is not playing nice with my HA.  I'm getting this..

> 
> Logger: homeassistant
> Source: custom_components/alarmdotcom/base_device.py:97
> Integration: Alarm.com ([documentation](https://www.github.com/uvjustin/alarmdotcom), [issues](https://www.github.com/uvjustin/alarmdotcom/issues))
> First occurred: 7:22:56 AM (200 occurrences)
> Last logged: 9:02:26 AM
> 
> Error doing job: Task exception was never retrieved
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 151, in _handle_refresh_interval
>     await self._async_refresh(log_failures=True, scheduled=True)
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
>     self.async_update_listeners()
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 110, in async_update_listeners
>     update_callback()
>   File "/config/custom_components/alarmdotcom/base_device.py", line 160, in _handle_coordinator_update
>     super()._handle_coordinator_update()
>   File "/config/custom_components/alarmdotcom/base_device.py", line 97, in _handle_coordinator_update
>     "raw_state_text": self._device.raw_state_text,
> AttributeError: 'NoneType' object has no attribute 'raw_state_text'

**Comment by Anto79-ops** [comment]
Update to the new beta release, it will fix it. Had the same issue but now it's fixed.

**Comment by elahd** [comment]
Thanks, guys. This is now released as v0.2.14.

@lutzky: The name change is this integration nudging in the direction of complying with the developer bulletin, [Adopting a new way to name entities](https://developers.home-assistant.io/blog/2022/07/10/entity_naming/).
